File: leetcode/python/find-words-containing-character.py
#!/usr/bin/env python3

""" Leetcode problem 2942: Find Words Containing Character. """

from typing import List


class Solution:
    def findWordsContaining(self, words: List[str], x: str) -> List[int]:
        result = []
        for i in range(0, len(words)):
            if x in set(words[i]):
                result.append(i)
        return result


def main() -> None:
    pass


if __name__ == "__main__":
    main()

File: leetcode/rust/find-words-containing-character.rs
// Leetcode problem 2942: Find Words Containing Character.

struct Solution {}

impl Solution {
    pub fn find_words_containing(words: Vec<String>, x: char) -> Vec<i32> {
        words
            .iter()
            .enumerate()
            .filter_map(|(i, word)| word.contains(x).then_some(i as i32))
            .collect()
    }
}

fn main() {}
